flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -411,6 +411,23 @@ class TaskManager(
             log.debug(s"Cannot find task to fail for execution ${executionID})")
           }
 
+        // stops a task
+        case StopTask(executionID) =>
+          val task = runningTasks.get(executionID)
+          if (task != null) {
+            try {
+              task.stopExecution()
+              sender ! new TaskOperationResult(executionID, true)
+            } catch {
+              case t: Throwable =>
+                        sender ! new TaskOperationResult(executionID, false, t.getMessage())
```

Comment:
Sure. That makes sense. But it does not answer my question. Right now, on error `StoppingFailure`, `CancelFailure`, and `TaskOperationResult` are sent back as messages. You started a discussion about removing `StoppingFailure` and `CancelFailure` and directly throw an exception. If this is the way to go, should `TaskOperationResult` be removed for the failure case, too? Or should `TaskOperationResult` removed for the failure case in any case, ie, independent from the `StoppingFailure` and `CancelFailure` discussion. If `TaskOperationResult` is kept, there is no way to get the exception back, as `TaskOperationResult` cannot take an exception as argument (or you change ``TaskOperationResult` -- but this should be an own JIRA, IMHO).